iothub_client/src/iothub_client_ll_uploadtoblob.c

```diff
@@ -538,15 +538,15 @@ static int IoTHubClient_LL_UploadToBlob_step1and2(IOTHUB_CLIENT_LL_UPLOADTOBLOB_
 }
 
 /*returns 0 when the IoTHub has been informed about the file upload status*/
-static int IoTHubClient_LL_UploadToBlob_step3(IOTHUB_CLIENT_LL_UPLOADTOBLOB_HANDLE_DATA* upload_data, STRING_HANDLE correlationId, HTTPAPIEX_HANDLE iotHubHttpApiExHandle, HTTP_HEADERS_HANDLE requestHttpHeaders, BUFFER_HANDLE messageBody)
+static int IoTHubClient_LL_UploadToBlob_step3(IOTHUB_CLIENT_LL_UPLOADTOBLOB_HANDLE_DATA* upload_data, HTTPAPIEX_HANDLE iotHubHttpApiExHandle, HTTP_HEADERS_HANDLE requestHttpHeaders, BUFFER_HANDLE messageBody)
 {
     int result;
     /*here is step 3. depending on the outcome of step 2 it needs to inform IoTHub about the file upload status*/
     /*if step 1 failed, there's nothing that step 3 needs to report.*/
     /*this POST "tries" to happen*/
 
     /*Codes_SRS_IOTHUBCLIENT_LL_02_085: [ IoTHubClient_LL_UploadMultipleBlocksToBlob(Ex) shall use the same authorization as step 1. to prepare and perform a HTTP request with the following parameters: ]*/
-    STRING_HANDLE relativePathNotification = STRING_construct_sprintf("/devices/%s/files/notifications/%s%s", upload_data->deviceId, STRING_c_str(correlationId), API_VERSION);
```

Comment:
No you can send the correlation ID in the url or the body. We had the URL but the problem was if someone had a filename that was really long in the URL, it would cause issues when the correlation was also there. So to give more room for the file name, we just put the correlation ID in the body. Thats also the way that the documentation [here](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-file-upload) states to do it.